### PR TITLE
feat(annotations): Handle mode change events

### PR DESCRIPTION
**Pull request by ConradJChan**

### CODE REVIEW

**Review by mxiao6 — DISMISSED** [review]

File: src/lib/AnnotationControls.ts [hunk 1]
```diff
@@ -141,11 +141,26 @@ export default class AnnotationControls {
     };
 
     /**
-     * Region comment button click handler
+     * Update the mode. If the mode is different from what is currently saved in state,
+     * then reset the current controls and apply the active state based on the provided mode.
+     * @param mode Annotation mode
```
Comment (mxiao6) [inline src/lib/AnnotationControls.ts:146]: We decided to not use jsdoc in this file

Comment (ConradJChan) [reply src/lib/AnnotationControls.ts:146]: Gotcha thanks

**Review by mxiao6 — DISMISSED** [review]

File: src/lib/AnnotationControls.ts [hunk 2]
```diff
@@ -141,11 +141,23 @@ export default class AnnotationControls {
     };
 
     /**
-     * Region comment button click handler
+     * Update the mode. If the mode is different from what is currently saved in state,
+     * then reset the current controls and apply the active state based on the provided mode.
+     */
+    public updateMode(mode: AnnotationMode): void {
+        // Only update buttons if mode has changed
+        if (this.currentMode !== mode) {
```
Comment (jstoffan) [inline src/lib/AnnotationControls.ts:149]: Seems like this should be a guard clause.

**Review by jstoffan — DISMISSED** [review]

File: src/lib/AnnotationControls.ts [hunk 3]
```diff
@@ -141,11 +141,23 @@ export default class AnnotationControls {
     };
 
     /**
-     * Region comment button click handler
+     * Update the mode. If the mode is different from what is currently saved in state,
+     * then reset the current controls and apply the active state based on the provided mode.
+     */
+    public updateMode(mode: AnnotationMode): void {
```
Comment (jstoffan) [inline src/lib/AnnotationControls.ts:147]: `setMode`?

**Pull request merged**